fetching/pick-fetcher/src/index.ts

```diff
@@ -24,9 +24,9 @@ export function pickFetcher (fetcherByHostingType: Partial<Fetchers>, resolution
 }
 
 export function isGitHostedPkgUrl (url: string): boolean {
-  return (
+  return ((
     url.startsWith('https://codeload.github.com/') ||
     url.startsWith('https://bitbucket.org/') ||
     url.startsWith('https://gitlab.com/')
-  ) && url.includes('tar.gz')
+  ) && url.includes('tar.gz')) || url.startsWith('https://pkg.pr.new/')
```

Comment:
Can you add a comment to explain what pr.new is? Or make a function with a descriptive name.

Also, in what state is a package stored at pkg.pr.new? Is it a compiled version of the package? If yes, then I don't think we can consider it a git-hosted package.

Reply:
For example, the dependency `@vue/runtime-vapor` installed by `pnpm i https://pkg.pr.new/@vue/runtime-vapor@13520` mentioned earlier is one of the branches of the vue code. This package has not been released yet. Therefore, executing `pnpm patch @vue/runtime-vapor` will result in the following error.

![image](https://github.com/user-attachments/assets/ec5dd9df-3218-4a32-a990-70edf1d004ca)

Reply:
@zkochan Do you think it is feasible to handle dependencies installed via pkg.pr.new in this way? Or should we try another approach?